app/models/post.rb

```diff
@@ -1127,6 +1127,10 @@ def cannot_permanently_delete_reason(user)
     end
   end
 
+  def self.show_welcome_topic_banner?
```

Comment:
This method doesn't feel like it is the concern of the `Post` model but rather the `Site` model.